File: ISA_BACKEND/BloodBank/src/main/java/com/example/bloodbank/Model/BloodDonationAppointment.java
package com.example.bloodbank.Model;

import lombok.*;

import javax.persistence.*;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "BloodDonationAppointments")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class BloodDonationAppointment {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private long id;

    private LocalDateTime startDateTime;
    private int duration;

    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "medicineStaff_id", referencedColumnName = "id")
    private Set<MedicineStaff> medicineStaffs = new HashSet<MedicineStaff>();

    @ManyToOne()
    @JoinColumn(name = "bloodBank_id", referencedColumnName = "id")
    private BloodBank bloodBank;
    private boolean isFree;
}

File: ISA_FRONTEND/src/app/model/bloodDonationAppointment.ts
import { BloodBank } from "./bloodBank";
import { MedicineStaff } from "./medicineStaff";

export class BloodDonationAppointment {
    id: string = '';
    startDateTime: string = '';
    duration: number = 0;
    medicineStaffs: MedicineStaff[] = new Array();
    bloodBank: BloodBank = new BloodBank;

    public constructor(obj?: any) {
        if (obj) {
            this.id = obj.id;
            this.startDateTime = obj.startDateTime;
            this.duration = obj.duration;
            this.medicineStaffs = obj.medicineStaffs;
            this.bloodBank = obj.bloodBank;
        }
    }
}